js/src/common/utils/ItemList.ts

```diff
@@ -1,9 +1,18 @@
-class Item {
-  content: any;
+class Item<T> {
+  content: T;
   priority: number;
+
+  /**
+   * The index of this item in the latest generated array.
+   *
+   * Set when calling `.toArray()`.
+   *
+   * **NOTE:** If you modify the item list after `.toArray()` is called,
```

Comment:
I don't think we should consider this to be part of the public API. Do we even need to set this on the internal representation?

Reply:
We don't use it internally, no. There's a chance someone somewhere uses it, but if you think we should just remove it, I'm fine with that.

Reply:
Let's remove it.